### PR TITLE
[FIX] payment_authorize: use name to send values when partner is a company

### PR DESCRIPTION
**Pull request by fernandahf**
Description of the issue/feature this PR addresses:

Currently, the name fields to send to Authorize when the partner is a company are:

* `firstName`
* `lastName`

if we consider the following name `Company Duck Inc`:

the code is sending:

* firstName: ''
* lastName: 'Duck'

Only it sends the `lastName` with the second word found in the name, due to the new validations in Authorize.Net, this kind of transaction is marked as suspicious and it's not confirming the transactions, leave them as pending, this commit is sending the full name in `lastName` instead of only second word to meet with the validation in Authorize.Net, in fact in the previous version, Odoo had the behavior of this commit:

https://github.com/odoo/odoo/blob/8533c19a6020461081c952c74bd6e043d26c79c7/addons/payment_authorize/models/authorize_request.py#L157-L159
    
Also, we are sending the fields `firstName` and  `lastName`, with the maximum length allowed by Authorize, which is 50:

https://developer.authorize.net/api/reference/index.html

Screenshot 2023-02-14 at 16 14 54

Current behavior before PR:

Sending only the second word of the name in `lastName` to Authorize.Net when the partner is a company.

Desired behavior after PR is merged:

Sending full name in `lastName` to Authorize.Net when the partner is a company.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by fernandahf** [comment]
FYI @moylop260

**Review by moylop260 — APPROVED** [review]
👍

**Comment by moylop260** [comment]
Hi @mart-e 

Sorry, Who is the man here?

**Comment by mart-e** [comment]
> Sorry, Who is the man here?

@moylop260 we have code owners activeted now, so somebody from @odoo/rd-payment should have a look into it

**Review by Demesmaeker — APPROVED** [review]
Hello,
lgtm, thanks for your fix!

**Comment by AntoineVDV** [comment]
@robodoo delegate=Demesmaeker

**Comment by Demesmaeker** [comment]
@robodoo r+ =]

**Comment by Demesmaeker** [comment]
Hello @fernandahf 
Could you rebase your branch to today version? 
Thank you =]

**Comment by AntoineVDV** [comment]
@Demesmaeker I launched a new batch with automatic rebase on runbot to remove the need for rebasing the branch ;)

**Pull request closed without merging**
